sdl2promela/builtins.py

```diff
@@ -213,81 +214,40 @@ def translate_builtin(
         return __translate_num(call, parameters)
     elif call.name.lower() == "exist":
         return __translate_exist(call, parameters)
-    raise NotImplementedError(f"Built in {call.name} is not yet implemented")
+    raise NotImplementedError(f"Builtin {call.name} is not yet implemented")
 
 
 def translate_assignment(
-    inlineName: str,
+    inlineName: Optional[str],
     call: sdlmodel.ProcedureCall,
-    left: promelamodel.Expression,
+    left: promelamodel.VariableReference,
     parameters: List[promelamodel.Expression],
     types: List[Asn1Type],
     allTypes: Dict[str, Asn1Type],
-):
+) -> promelamodel.Statement:
     """
     Translate an assignment, where right side is built-in function.
-    :param inlineName: name of inline, which assigns value
+
+    :param inlineName: name of inline, which assigns value, or None when
+                       assignment shall be used
     :param call: Call to be translated.
     :param parameters: List of translated call parameters.
     :param types: List of types of parameters.
     :param allTypes: All available asn1 datatypes
-    :returns: Translation of the assignment.
+    :returns: promela statement, which is translated version of SDL assignment
     """
-    if call.name.lower() == "length":
-        return (
-            CallBuilder()
-            .withTarget(inlineName)
-            .withParameter(left)
-            .withParameter(__translate_length(call, parameters, types, allTypes))
-            .build()
-        )
-    if call.name.lower() == "present":
-        return (
-            CallBuilder()
-            .withTarget(inlineName)
-            .withParameter(left)
-            .withParameter(__translate_present(call, parameters))
-            .build()
+    right = translate_builtin(call, parameters, types, allTypes)
+    if right is None:
+        raise Exception(
+            f"Builtin {call.name} cannot be used on the left side of assignment"
```

Comment:
if right is none then left cannot be used?

Reply:
changed